incubator/hnc/internal/reconcilers/object.go

```diff
@@ -666,7 +665,10 @@ func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstruct
 	if err != nil {
 		log.Error(err, "Invalid value")
 	}
-	noneSelector := r.getNoneSelector(log, inst)
+	noneSelector, err := selectors.GetNoneSelector(inst)
+	if err != nil {
+		log.Error(err, "Invalid value")
```

Comment:
I've been trying to figure out what we should say here instead of "Invalid value" since that just duplicates what would be in `err`. But it took me three PRs to come up with something!

What this line of code "knows" that no-one else knows is that we're trying to decide if the source can be propagated or not. `selectors.GetXXXSelector` just knows that someone cares about the selector value but doesn't know what the caller wants to do with it. But on this line, we know that if there's an error, _we're not going to propagate it to anyone_.

So I think that's the message we should show in the log (here and also on the other two similar lines): something like "Cannot propagate" or "Will not propagate". That will make a lot more sense than "Invalid value" to a cluster admin user who sees this in their log.

Reply:
Agreed, thanks for the nice explanation!